### ISSUE TITLE
Future of the grid branch?

### ISSUE THREAD

**Issue by jjrv** [phosphorjs/phosphor#184]
Are there still plans to finish the canvas-based grid component? It seems like a very large but useful feature (sounds like more work than the whole rest of PhosphorJS, depending on supported features).

Currently I'm using PhosphorJS with [fin-hypergrid](https://github.com/openfin/fin-hypergrid), also canvas-based.

**Comment by sccolbert** [comment]
Yes. It will be completed this year. I will start back on that after I release the 1.0 packages.

**Comment by jjrv** [comment]
Some notes for it:

- `fin-hypergrid` has concluded that drawing borders per cell is too slow, they draw borders for entire rows and columns.
- If multiline text in cells is supported, a custom renderer needs to be able to specify a hook for measuring the cell height, before that row and rows below it can be rendered.
- Maps like Leaflet handle scrolling extremely fast by having multiple canvases and moving their DOM elements, instead of blitting within a canvas.

**Comment by sccolbert** [comment]
Thanks!

**Comment by jasongrout** [comment]
Chris, it would be great to have your thoughts on fin-hypergrid too.

**Comment by sccolbert** [comment]
I looked at it pretty seriously before.  I liked its pattern for handling cell renderers, and I've adopted that idea with slight modification.

A few things that I don't like about it:
- When it redraws, it redraws *everything* rather than just the dirty region.
- In order to handle resizing, it redraws everything on a polling timer.
- I'm not a fan of the code architecture in general, as there is tight coupling between components.
